### PR TITLE
Scheduler tutorial

### PR DESCRIPTION
**Pull request by lena-kashtelyan**
Complete OSS tutorial for the Ax scheduler

### CODE REVIEW

File: tutorials/scheduler.ipynb [hunk 1]
```diff
@@ -0,0 +1,1043 @@
...
+    "   2. Using `Scheduler.run_trials_and_yield_results` to run the optimization via a generator method."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1. `Scheduler` and external systems for trial evaluation\n",
+    "\n",
+    "`Scheduler` is a closed-loop manager class in Ax that continuously deploys trial runs to an arbitrary external system, polls their status from that system, and leverages known trial results to generate more trials.\n",
```
Comment (Balandat) [inline tutorials/scheduler.ipynb:32]:
```suggestion
    "`Scheduler` is a closed-loop manager class in Ax that continuously deploys trial runs to an arbitrary external system in an asynchronous fashion, polls their status from that system, and leverages known trial results to generate more trials.\n",
```

File: tutorials/scheduler.ipynb [hunk 2]
```diff
@@ -0,0 +1,1043 @@
...
+    "\n",
+    "Key features of the `Scheduler`:\n",
+    "- Maintains user-set concurrency limits for trials run in parallel, keep track of tolerated level of failed trial runs, and 'oversee' the optimization in other ways,\n",
+    "- Leverages an Ax `Experiment` for optimization setup (an optimization config with metrics, a search space, a runner for trial evaluations),\n",
+    "- Relies on an Ax `GenerationStrategy` for a specification of an optimization algorithm used to generate new trials to run,\n",
```
Comment (Balandat) [inline tutorials/scheduler.ipynb:37]:
```suggestion
    "- Uses an Ax `GenerationStrategy` for flexible specification of an optimization algorithm used to generate new trials to run,\n",
```

File: tutorials/scheduler.ipynb [hunk 3]
```diff
@@ -0,0 +1,1043 @@
...
+    "            # or to an actual value when SEM is known. Setting SEM to\n",
+    "            # `None` results in Ax assuming unknown noise and inferring\n",
+    "            # noise level from data.\n",
+    "            \"sem\": None,\n",
+    "        }\n",
+    "        return Data(df=pd.DataFrame.from_records([df_dict]))"
+   ]
+  },
+  {
+   "cell_type"
```
Comment (Balandat) [inline tutorials/scheduler.ipynb:293]:
```suggestion
    "            status_dict[status].add(trial.index)\n",
```

Comment (Balandat) [inline tutorials/scheduler.ipynb:284]:
```suggestion
    "        status_dict = defaultdict(set)\n",
```

Comment (Balandat) [reply tutorials/scheduler.ipynb:284]: ugh somehow the json messes up the code suggestion. You know what I mean, I hope.

Comment (Balandat) [inline tutorials/scheduler.ipynb:303]: let's add a link to the tutorial here?

Comment (Balandat) [inline tutorials/scheduler.ipynb:346]: This might be confusing for users. Is this something we can just do? Or not issue this info / make it debug level?

Comment (Balandat) [inline tutorials/scheduler.ipynb:716]:
```suggestion
    "To store state of optimization to an SQL backend, first follow the [setup instructions](https://ax.dev/docs/storage.html#sql) on the Ax website. Having set up the SQL backend, pass `DBSettings` to scheduler on instantiation (note that SQLAlchemy dependency will have to be installed – for installation, refer to [optional dependencies](https://ax.dev/docs/installation.html#optional-dependencies) on Ax website):"
```

Comment (Balandat) [inline tutorials/scheduler.ipynb:737]: ocalhost?

Comment (Balandat) [inline tutorials/scheduler.ipynb:795]:
```suggestion
    "With the newly reloaded experiment, the Scheduler can continue the optimization:"
```

Comment (Balandat) [inline tutorials/scheduler.ipynb:1018]:
```suggestion
    "    print(\"Reported result: \", reported_result)"
```

**Review by Balandat — APPROVED** [review]
This is great, I love how this is fully self-contained and doesn't need any external deps. Some minor comments inline. 

Something that would be nice to have (though not in this PR) in order to demonstrate this on an actual (albeit toy) problem would be a simple HPO example that implements a "backend system" that actually does some model fitting locally. We could just use a multiprocessing setup for this where a single trial could be to run some sklearn training job for as simple model.

File: tutorials/scheduler.ipynb (hunk 3 above)
Comment (lena-kashtelyan) [reply tutorials/scheduler.ipynb:284]: Yeah for sure! I was a bit worried that not everyone is familiar with `defaultdict` and felt like the overhead here was low enough to not use it, but I think you're right that anyone dealing with this tutorial is probably familiar with it : )

Comment (lena-kashtelyan) [reply tutorials/scheduler.ipynb:303]: Will do

Comment (lena-kashtelyan) [reply tutorials/scheduler.ipynb:346]: Yeah for sure. Let me log that as debug

Comment (Balandat) [reply tutorials/scheduler.ipynb:284]: If they aren't yet they should be, so teaching them is an added benefit of the tutorial :)

**Pull request closed without merging**